PSReadLine/PlatformWindows.cs

```diff
@@ -218,4 +291,17 @@ private static bool IsHandleRedirected(bool stdin)
         // Char device - if GetConsoleMode succeeds, we are NOT redirected.
         return !GetConsoleMode(handle, out var unused);
     }
+
+    public static bool IsConsoleApiAvailable()
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return false;
+        }
+        if (IsHandleRedirected(stdin: true) && IsHandleRedirected(stdin: false))
```

Comment:
@parkovski - is this line correct? It seems like redirected stdout is enough to be a problem, see #591.

Reply:
It should probably have an input/output argument like `IsHandleRedirected`. I think there are four cases for console APIs:
- Input is not redirected; input APIs are usable.
- Output is not redirected; output APIs are usable.
- At least one of input or output is not redirected; things like `SetConsoleTitle` are usable.
- Both streams are redirected; we can't use any console APIs.

It's only used for the first scenario right now, but in case any of the others are needed, my new PR fixes it to work like this.